### ISSUE TITLE
COMPAT: force int when doing unit conversions in pd.to_datetime

### ISSUE THREAD

**Issue by jreback** [pydata/pandas#11760]
from [SO](http://stackoverflow.com/questions/34078114/convert-pandas-time-series-unix-epoch-to-datetime/34078557#34078557)
xref to #11758 

We should force int conversions when ``unit`` is passed (as by definition these are int-like things).
```
In [21]: s = pd.Series(["1440643875", "1440644191", "1440645638", "1440998720"], dtype=object)

In [22]: pd.to_datetime(s,unit='s')
ValueError: year is out of range

In [23]: pd.to_datetime(s.astype(int),unit='s')
Out[23]: 
0   2015-08-27 02:51:15
1   2015-08-27 02:56:31
2   2015-08-27 03:20:38
3   2015-08-31 05:25:20
dtype: datetime64[ns]
```

**Comment by gliptak** [comment]
This seem to have been corrected in current HEAD:

```
In [5]: s = pd.Series(["1440643875", "1440644191", "1440645638", "1440998720"], dtype=object)

In [6]: pd.to_datetime(s,unit='s')
Out[6]: 
0   2015-08-27 02:51:15
1   2015-08-27 02:56:31
2   2015-08-27 03:20:38
3   2015-08-31 05:25:20
dtype: datetime64[ns]
```

**Comment by jreback** [comment]
this was closed by #13003 and is already sufficiently tested.
